packages/dataviews/src/types.ts

```diff
@@ -3,21 +3,49 @@
  */
 import type { ReactNode } from 'react';
 
-interface Option {
-	value: any;
+export type SortDirection = 'asc' | 'desc';
+
+/**
+ * Type of view UI.
+ */
+export type ViewType = 'table' | 'grid' | 'list';
+
+/**
+ * Type of field.
+ */
+export type FieldType = 'enumeration';
+
+/**
+ * Generic option type.
+ */
+interface Option< Value extends any = any > {
+	value: Value;
 	label: string;
 }
 
-interface filterByConfig {
+interface FilterByConfig {
+	/**
+	 * The list of operators supported by the field.
+	 */
 	operators?: Operator[];
+
+	/**
+	 * Whether it is a primary filter.
+	 *
+	 * A primary filter is always visible and is not listed in the "Add filter" component,
+	 * except for the list layout where it behaves like a secondary filter.
+	 */
 	isPrimary?: boolean;
 }
 
 type Operator = 'is' | 'isNot' | 'isAny' | 'isNone' | 'isAll' | 'isNotAll';
 
 export type Item = Record< string, any >;
```

Comment:
It would be nice to rename this `AnyItem`, then all the generic arguments could be `Item`, otherwise it conflicts with this declaration of `Item`.

Reply:
Appending an `Any` to a type used base for a generic argument type is a common in the `@wordpress/data` package.

Reply:
Let's do that.

Reply:
Updated. I also just captured the addition of `bulk-actions.tsx`